nemo/collections/diffusion/data/diffusion_taskencoder.py

```diff
@@ -0,0 +1,121 @@
+            - All keys from the result of `basic_sample_keys(sample)`
+            - 'json': The contains meta data like resolution, aspect ratio, fps, etc.
+            - 'pth': contains video latent tensor
+            - 'pickle': contains text embeddings
+    """
+    return dict(
+        **basic_sample_keys(sample),
+        json=sample['.json'],
+        pth=sample['.pth'],
+        pickle=sample['.pickle'],
+    )
+
+
+class BasicDiffusionTaskEncoder(DefaultTaskEncoder, IOMixin):
+    """
+    BasicDiffusionTaskEncoder is a class that encodes image/video samples for diffusion tasks.
+    Attributes:
+        cookers (list): A list of Cooker objects used for processing.
+        max_frames (int, optional): The maximum number of frames to consider from the video. Defaults to None.
+        text_embedding_padding_size (int): The padding size for text embeddings. Defaults to 512.
+    Methods:
+        __init__(*args, max_frames=None, text_embedding_padding_size=512, **kwargs):
+            Initializes the BasicDiffusionTaskEncoder with optional maximum frames and text embedding padding size.
+        encode_sample(sample: dict) -> dict:
+            Encodes a given sample dictionary containing video and text data.
+            Args:
+                sample (dict): A dictionary containing 'pth' for video latent and 'json' for additional info.
+            Returns:
+                dict: A dictionary containing encoded video, text embeddings, text mask, and loss mask.
+            Raises:
+                SkipSample: If the video latent contains NaNs, Infs, or is not divisible by the tensor parallel size.
+    """
+
+    cookers = [
+        Cooker(cook),
+    ]
+
+    def __init__(self, *args, max_frames: int = None, text_embedding_padding_size: int = 512, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.max_frames = max_frames
+        self.text_embedding_padding_size = text_embedding_padding_size
+
+    def encode_sample(self, sample: dict) -> dict:
+        video_latent = sample['pth']
+
+        if torch.isnan(video_latent).any() or torch.isinf(video_latent).any():
+            raise SkipSample()
+        if torch.max(torch.abs(video_latent)) > 1e3:
+            raise SkipSample()
+
+        info = sample['json']
+        _, T, H, W = video_latent.shape
+        is_image = T == 1
+
+        if self.max_frames is not None:
+            video_latent = video_latent[:, : self.max_frames, :, :]
+
+        tpcp_size = parallel_state.get_tensor_model_parallel_world_size()
+        if parallel_state.get_context_parallel_world_size() > 1:
+            tpcp_size *= parallel_state.get_context_parallel_world_size() * 2
+        if (T * H * W) % tpcp_size != 0:
+            print(f'skipping {video_latent.shape=} not divisible by {tpcp_size=}')
+            raise SkipSample()
+
+        seq_len = video_latent.shape[-1] * video_latent.shape[-2] * video_latent.shape[-3]
```

Comment:
Just FYI, when training Neva model, we noticed that using dynamic sequence length is causing slow down in the model forward pass (almost 3x). So for the other task encoders I changed it so that we always pad to some predefined. max sequence length

Reply:
makes sense. we're going to use packing and will pad after packing